pennylane/ops/functions/equal.py

```diff
@@ -99,3 +124,35 @@ def equal(
                 return False
 
     return getattr(op1, "inverse", False) == getattr(op2, "inverse", False)
+
+
+def equal_measurements(op1, op2):
+    """Determine whether two MeasurementProcess objects are equal"""
+    return_types_match = op1.return_type == op2.return_type
+    if op1.obs is not None and op2.obs is not None:
+        observables_match = equal(op1.obs, op2.obs)
+    # check obs equality when either one is None (False) or both are None (True)
+    else:
+        observables_match = op1.obs == op2.obs
+    wires_match = op1.wires == op2.wires
+    eigvals_match = qml.math.allequal(op1.eigvals(), op2.eigvals())
+    log_base_match = op1.log_base == op2.log_base
+
+    return (
+        return_types_match
+        and observables_match
+        and wires_match
+        and eigvals_match
+        and log_base_match
+    )
+
+
+def equal_shadow_measurements(op1, op2):
+    """Determine whether two ShadowMeasurementProcess objects are equal"""
+    return_types_match = op1.return_type == op2.return_type
+    wires_match = op1.wires == op2.wires
+    H_match = op1.H == op2.H
+    k_match = op1.k == op2.k
+    seed_class_match = op1.seed.__class__ == op2.seed.__class__  # seed can be None or a random int
```

Comment:
Does this make sense? we wouldn't want to check if `op1.seed == op2.seed` since it's either a random number or None, so im only checking the `__class__`.

Reply:
`seed` is never `None`, given that the `ShadowMeasurementProcess` class is called in the `classical_shadow` and `shadow_expval` functions, and both define a `seed` value.

I would delete this line. We still want to compare measurements even if they are stochastic. For example: `qml.equal(qml.counts(wires=0), qml.counts(wires=0))` should return `True`, even though this measurement will give different results for different executions (because it depends on random processes).

Reply:
> We still want to compare measurements even if they are stochastic

This makes sense!

However, the way both functions define the `seed` value is: `seed = np.random.randint(2**30) if seed_recipes else None`
as shown [here](https://github.com/PennyLaneAI/pennylane/blob/master/pennylane/measurements.py#L1489).
So, if `seed_recipes=False`, then it would be None.

Is there something that I am missing? You likely have more visibility into the project than I do, so your insight is appreciated.

Reply:
My mistake. You are completely right, `seed` can be `None`.

However we still don't want to compare the seed inside `qml.equal`.